### PR TITLE
Feature/unit tests

### PR DESCRIPTION
**Pull request by firethar**
Add: mockServices

### CODE REVIEW

File: src/app/product.service.spec.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
 import { TestBed } from '@angular/core/testing';
 import { SoldProducts } from './product';
 
 import { ProductService } from './product.service';
 
 describe('ProductService', () => {
   let service: ProductService;
+  let httpTestingController: HttpTestingController
+  const URL = 'https://private-anon-cfb1043af1-swoproducts.apiary-mock.com/soldProducts';
```
Comment (michalzielonka) [inline src/app/product.service.spec.ts]: Isn't it the same as in env? Can you take it from env file?

File: src/app/product.service.spec.ts [hunk 2]
```diff
@@ -24,23 +26,34 @@ describe('ProductService', () => {
...
     service = TestBed.inject(ProductService);
   });
-
+  
   it('should be created', () => {
     expect(service).toBeTruthy();
   });
+  
+  it('should get productsData$ from URL', () => {
+    service.productsData$.subscribe();
```
Comment (michalzielonka) [inline src/app/product.service.spec.ts]: Validate methoda calls inside subscribe

File: src/app/sold-products/sold-products.component.spec.ts [hunk 3]
```diff
@@ -1,43 +1,61 @@
...
-  let service: ProductService;
+  let mockProductsService: ProductService;
+  let SOLD_PRODUCTS: SoldProducts; 
 
-  beforeEach(async () => {
-    await TestBed.configureTestingModule({
-      imports: [
-        HttpClientModule,
+  beforeEach( () => {
+    SOLD_PRODUCTS = {
```
Comment (michalzielonka) [inline src/app/sold-products/sold-products.component.spec.ts]: Keep mocks in separate file. It is easier to modify and read

File: src/app/sold-products/sold-products.component.spec.ts [hunk 4]
```diff
@@ -1,43 +1,61 @@
...
-    service = TestBed.inject(ProductService);
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
  
-  it('service should be injected', () => {
-    expect(service).toBeTruthy();
+  it('have injected service', () => {
```
Comment (michalzielonka) [inline src/app/sold-products/sold-products.component.spec.ts]: I don't think you need this test :)

**Review by michalzielonka — CHANGES_REQUESTED** [review]
Looks good :) Minor comments 👍

File: src/app/product.service.spec.ts (hunk 1 above)
Comment (firethar) [reply src/app/product.service.spec.ts]: Indeed. Updated to env.baseURL

File: src/app/product.service.spec.ts (hunk 2 above)
Comment (firethar) [reply src/app/product.service.spec.ts]: What do you mean?

File: src/app/sold-products/sold-products.component.spec.ts (hunk 4 above)
Comment (firethar) [reply src/app/sold-products/sold-products.component.spec.ts]: Ok :)

File: src/app/sold-products/sold-products.component.spec.ts (hunk 3 above)
Comment (firethar) [reply src/app/sold-products/sold-products.component.spec.ts]: Updated

File: src/app/product.service.spec.ts (hunk 2 above)
Comment (michalzielonka) [reply src/app/product.service.spec.ts]: service.productsData$.subscribe({
put asserts here :)
});

Comment (firethar) [reply src/app/product.service.spec.ts]: Updated

**Review by michalzielonka — APPROVED** [review]

**Pull request merged**